src/Storage/Field/Type/FieldTypeBase.php

```diff
@@ -97,4 +144,21 @@ public function getStorageType()
     {
         return Type::getType($this->mapping['type']);
     }
+
+    /**
+     * Check if a value is a JSON string.
+     *
+     * @param mixed $value
+     *
+     * @return boolean
+     */
+    protected function isJson($value)
+    {
+        if (!is_string($value)) {
+            return false;
+        }
+        json_decode($value);
```

Comment:
You're expecting PHP to act rationally? `json_decode()` just fails silently and you need to check `json_last_error()`… hence the reason `json_last_error()` is called belonw :stuck_out_tongue_winking_eye: